charts/pega/config/deploy/context.xml.tmpl

```diff
@@ -21,20 +21,20 @@
     minEvictableIdleTimeMillis="60000"
     />
 
-  {{ if and .Env.DB_RW_USERNAME .Env.JDBC_RW_URL }}
+  {{ if or .Env.SET_RW .Env.JDBC_RW_URL }}
     <Resource name="jdbc/PegaRULESLongRW"
     auth="Container"
     type="javax.sql.DataSource"
     driverClassName="{{ .Env.JDBC_CLASS }}"
-    url="{{ .Env.JDBC_RW_URL }}"
-    username="{{ .Env.DB_RW_USERNAME }}"
-    password=""
+    url="{{- if .Env.JDBC_RW_URL -}}{{ .Env.JDBC_RW_URL }}{{- else -}}{{ .Env.JDBC_URL }}{{- end -}}"
+    username="{{ .Env.SECRET_DB_USERNAME }}"
+    password="{{ .Env.SECRET_DB_PASSWORD }}"
     maxTotal="{{ .Env.JDBC_MAX_ACTIVE }}"
-    minIdle="{{ default .Env.JDBC_RO_MIN_IDLE .Env.JDBC_MIN_IDLE }}"
+    minIdle="{{ .Env.JDBC_MIN_IDLE }}"
     maxIdle="{{ .Env.JDBC_MAX_IDLE }}"
     maxWaitMillis="{{ .Env.JDBC_MAX_WAIT }}"
-    initialSize="{{ default .Env.JDBC_RO_INITIAL_SIZE .Env.JDBC_INITIAL_SIZE }}"
-    connectionProperties="socketTimeout=1200;statement_timeout=1200500;idle_in_transaction_session_timeout=1200500;"
+    initialSize="{{ .Env.JDBC_INITIAL_SIZE }}"
+    connectionProperties="{{ .Env.JDBC_CONNECTION_PROPERTIES }}socketTimeout=1200;statement_timeout=1200500;idle_in_transaction_session_timeout=1200500;"
```

Comment:
Do we want to rely on there being a trailing semicolon in the user-supplied props?  Not sure that's a good idea.

Reply:
Fair enough. Would starting with a semicolon be a problem if there are no connection properties passed though? Guessing probably not, it would just count as a null/nil/empty connection property, just want to confirm